core/src/main/java/org/apache/brooklyn/core/effector/ProxyEffector.java

```diff
@@ -0,0 +1,99 @@
+import org.apache.brooklyn.core.config.ConfigKeys;
+import org.apache.brooklyn.core.effector.Effectors.EffectorBuilder;
+import org.apache.brooklyn.core.mgmt.internal.EffectorUtils;
+import org.apache.brooklyn.util.core.config.ConfigBag;
+import org.apache.brooklyn.util.core.task.Tasks;
+import org.apache.brooklyn.util.guava.Maybe;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+
+public class ProxyEffector extends AddEffector {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ProxyEffector.class);
+
+    public static final ConfigKey<Entity> TARGET_ENTITY = ConfigKeys.newConfigKey(Entity.class,
+            "targetEntity", "The proxy target");
+
+    public static final ConfigKey<String> TARGET_EFFECTOR_NAME = ConfigKeys.newStringConfigKey(
+            "targetEffector", "The effector to invoke on the target entity");
+
+    public ProxyEffector(ConfigBag params) {
+        super(newEffectorBuilder(params).build());
+    }
+
+    public ProxyEffector(Map<?, ?> params) {
+        this(ConfigBag.newInstance(params));
+    }
+
+    public static EffectorBuilder<Object> newEffectorBuilder(ConfigBag params) {
+        EffectorBuilder<Object> eff = AddEffector.newEffectorBuilder(Object.class, params);
+        eff.impl(new Body(eff.buildAbstract(), params));
+        return eff;
+    }
+
+    protected static class Body extends EffectorBody<Object> {
+
+        private final Effector<?> effector;
+        private final Object proxy;
+        private final String effectorName;
+
+        public Body(Effector<?> eff, ConfigBag params) {
+            this.effector = eff;
+            // Don't use getConfig(TARGET_ENTITY) because DslComponents can't be
+            // coerced to entities at this point.
+            this.proxy = Preconditions.checkNotNull(params.getAllConfigRaw().get(TARGET_ENTITY.getName()),
+                    "Target entity must be supplied when defining this effector");
+            this.effectorName = Preconditions.checkNotNull(params.get(TARGET_EFFECTOR_NAME), "Target effector name must be supplied when defining this effector");
+        }
+
+        @Override
+        public Object call(ConfigBag params) {
+            Entity target = resolveTarget().get();
+            return invokeEffectorNamed(target, effectorName, params);
+        }
+
+        private Maybe<Entity> resolveTarget() {
+            return Tasks.resolving(proxy, Entity.class)
+                    .context(entity())
+                    .swallowExceptions()
+                    .getMaybe();
+        }
+
+        private Object invokeEffectorNamed(Entity target, String effectorName, ConfigBag params) {
+            LOG.debug("{} forwarding effector invocation on {} to entity={}, effector={}, paramters={}",
+                    new Object[]{this, entity(), target, effectorName, params});
+            Effector<?> effector = EffectorUtils.findEffectorDeclared(target, effectorName).get();
```

Comment:
Yes, but will the exception have enough clear info in the message to explain what's gone wrong?  Maybe so, which is fine, but I was thinking just that if we construct the message in this routine we can add info about effector name, target, perhaps parameters.